src/Steeltoe.Discovery.EurekaBase/EurekaDiscoveryClientHandlerProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Net.Http;
+
+namespace Steeltoe.Discovery.Eureka
+{
+    public class EurekaDiscoveryClientHandlerProvider
```

Comment:
I'm thinking an interface (e.g. IDiscoveryClientHandlerProvider) like below added to this class. 

public interface IDiscoveryClientHandlerProvider {
    HttpClientHandler GetHttpClientHandler();
}

Then change all constructors to inject/use the interface, instead of the class.

Reply:
changed as requested